packages/stark-ui/src/modules/keyboard-directives/directives/transform-input.directive.spec.ts

```diff
@@ -0,0 +1,111 @@
+import { Component, OnInit } from "@angular/core";
+import { ComponentFixture, TestBed } from "@angular/core/testing";
+import { FormControl, FormsModule, ReactiveFormsModule } from "@angular/forms";
+import { By } from "@angular/platform-browser";
+import { MockStarkLoggingService } from "@nationalbankbelgium/stark-core/testing";
+import { STARK_LOGGING_SERVICE } from "@nationalbankbelgium/stark-core";
+import { StarkTransformInputDirective } from "./transform-input.directive";
+
+/**
+ * @ignore
+ * Mocks an InputEvent on the element with the given value.
+ * @param value - the value to set the element to.
+ * @param element - the HTMLInputElement to mock the event on
+ */
+const mockInputEvent: (value: string, element: HTMLInputElement) => void = (value: string, element: HTMLInputElement): void => {
+	const event: Event = document.createEvent("Event");
+	event.initEvent("input", true, true);
+	element.value = value;
+	element.dispatchEvent(event);
+};
+
+describe("TransformsInputDirective with ngModel", () => {
```

Comment:
I propose to change the structure of this file as follows so we cover all use cases:

```
describe("TransformsInputDirective") {
    describe("ngModel")
        // 'it' blocks to test "lowercase", "uppercase", a dummy transform function and an invalid transformation input or undefined
    describe("formControl")
        // 'it' blocks to test "lowercase", "uppercase", a dummy transform function and an invalid transformation input or undefined
}
```

So maybe it is a good idea to extract the tests to a function so you can re-use them in both scenarios: ngModel and FormControl.

What do you think?

Reply:
I agree about writing some more test-cases and bundeling them like you show. But I think keeping tests atomic is better and would prefer duplicating some code.